x-pack/plugins/apm/server/routes/fleet.ts

```diff
@@ -31,6 +39,144 @@ const hasFleetDataRoute = createApmServerRoute({
+    const { attributes } = await savedObjectsClient.get(
+      APM_SERVER_SCHEMA_SAVED_OBJECT_TYPE,
+      APM_SERVER_SCHEMA_SAVED_OBJECT_ID
+    );
+    return attributes;
+  },
+});
+
+const getUnsupportedApmServerSchemaRoute = createApmServerRoute({
+  endpoint: 'GET /api/apm/fleet/apm_server_schema/unsupported',
+  options: { tags: ['access:apm'] },
+  handler: async (resources) => {
+    const { context } = resources;
+    const savedObjectsClient = context.core.savedObjects.client;
+    return {
+      unsupported: await getUnsupportedApmServerSchema({ savedObjectsClient }),
+    };
+  },
+});
+
+const getMigrationCheckRoute = createApmServerRoute({
+  endpoint: 'GET /api/apm/fleet/migration_check',
+  options: { tags: ['access:apm'] },
+  handler: async (resources) => {
+    const { plugins, context, config, request } = resources;
+    const cloudApmMigrationEnabled =
+      config['xpack.apm.agent.migrations.enabled'];
+    if (!plugins.fleet || !plugins.security) {
+      throw Boom.internal(fleetSecurityPluginsRequireMessage);
+    }
+    const savedObjectsClient = context.core.savedObjects.client;
+    const fleetPluginStart = await plugins.fleet.start();
+    const securityPluginStart = await plugins.security.start();
+    const hasRequiredRole = isSuperuser({ securityPluginStart, request });
+    const cloudApmPackagePolicy = await getCloudAgentPolicy({
+      savedObjectsClient,
+      fleetPluginStart,
+    });
+    if (!cloudApmPackagePolicy) {
+      return {
+        has_cloud_agent_policy: false,
+        has_cloud_apm_package_policy: false,
+        cloud_apm_migration_enabled: cloudApmMigrationEnabled,
+        has_required_role: hasRequiredRole,
+      };
+    }
+    const apmPackagePolicy = getApmPackagePolicy(cloudApmPackagePolicy);
+    return {
+      has_cloud_agent_policy: true,
+      has_cloud_apm_package_policy: !!apmPackagePolicy,
+      cloud_apm_migration_enabled: cloudApmMigrationEnabled,
+      has_required_role: hasRequiredRole,
+    };
+  },
+});
+
+const createCloudApmPackagePolicyRoute = createApmServerRoute({
+  endpoint: 'POST /api/apm/fleet/cloud_apm_package_policy',
+  options: { tags: ['access:apm', 'access:apm_write'] },
+  handler: async (resources) => {
+    const { plugins, context, config, request, logger } = resources;
+    const cloudApmMigrationEnabled =
+      config['xpack.apm.agent.migrations.enabled'];
+    if (!plugins.fleet || !plugins.security) {
+      throw Boom.internal(fleetSecurityPluginsRequireMessage);
+    }
+    const savedObjectsClient = context.core.savedObjects.client;
+    const coreStart = await resources.core.start();
+    const esClient = coreStart.elasticsearch.client.asScoped(resources.request)
+      .asCurrentUser;
+    const fleetPluginStart = await plugins.fleet.start();
+    const securityPluginStart = await plugins.security.start();
+    const hasRequiredRole = isSuperuser({ securityPluginStart, request });
```

Comment:
I explicitly check for superuser role here because in development, i was able to create the apm package policy and add it to the cloud policy as a user without the superuser role.